.github/workflows/update-dependency-deployments.yml

```diff
@@ -28,8 +27,8 @@ jobs:
         credentials: ${{ secrets.GKE_SA_KEY }}
 
     - name: Apply
-      run: ./k8s/deploy-collector.sh
-
-
-
-
+      run: |
+        chmod +x ./k8s/deploy-collector.sh
```

Comment:
You don't need this. Git already tracks permissions, so if you commit the file with 755 it will be executable